Sources/Display.php

```diff
@@ -851,26 +852,26 @@ function Display()
 		if (isset($_SESSION['page_next_start']) && $_SESSION['page_next_start'] == $start)
 		{
 			$start_char = 'M';
-			$page_id = $_SESSION['page_last_id'];
+			$page_id = $ascending ? $_SESSION['page_last_id'] : $_SESSION['page_first_id'];
 		}
 		// User moved to the previous page
 		elseif (isset($_SESSION['page_before_start']) && $_SESSION['page_before_start'] == $start)
 		{
 			$start_char = 'L';
-			$page_id = $_SESSION['page_first_id'];
+			$page_id = $ascending ? $_SESSION['page_first_id'] : $_SESSION['page_last_id'];
 		}
 		// User refreshed the current page
 		elseif (isset($_SESSION['page_current_start']) && $_SESSION['page_current_start'] == $start)
 		{
 			$start_char = 'C';
-			$page_id = $_SESSION['page_first_id'];
+			$page_id = $ascending ? $context['topicinfo']['id_fist_msg'] : $context['topicinfo']['id_last_msg'];
```

Comment:
Typo: `id_fist_msg` should be `id_first_msg`

Reply:
Also, why aren't you using the `$_SESSION['page_first_id']` and `$_SESSION['page_last_id']` values in this case?